digitaltwins/model_repository/src/cl/model_repository_service_client.ts

```diff
@@ -0,0 +1,332 @@
+   * @param {string} modelId                  The Id of the requested model.
+   * @param {GetModelParams} options          Optional argument with the following members:
+   *                                          JSON format:
+   *                                          options = {
+   *                                            'repositoryId': '',
+   *                                            'xMsClientRequestId': '',
+   *                                            'expand'
+   *                                          };
+   *                                          Where:
+   *                                          'repositoryId'
+   *                                              {string} Private repository id. To access global repository, caller should not specify this value.
+   *                                          'xMsClientRequestId'
+   *                                              {string} Provides a client-generated opaque value that is recorded in the logs.
+   *                                              Using this header is highly recommended for correlating client-side activities
+   *                                              with requests received by the server.
+   *                                          'expand'
+   *                                              {boolean} Indicates whether to expand the capability
+   *                                              model's interface definitions inline or not. This query parameter ONLY
+   *                                              applies to Capability model. Default value: false .
+   * @returns GetModelResponse                The return object containing the Model plus the HttpResponse.
+   * @memberof ModelRepositoryServiceClient
+   */
+  getModel(modelId: string, options?: GetModelParams): Promise<GetModelResponse>;
+  getModel(modelId: string, options?: GetModelParams, callback?: TripleValueCallback<GetModelResponse, msRest.HttpOperationResponse>): void;
+  getModel(modelId: string, options?: GetModelParams, callback?: TripleValueCallback<GetModelResponse, msRest.HttpOperationResponse>): void | Promise<GetModelResponse> {
+    /* Codes_SRS_NODE_MODEL_REPOSITORY_SERVICE_CLIENT_12_002: [ The `getModel` method shall call the `getModel` method of the protocol layer with the given arguments. ]*/
+    /* Codes_SRS_NODE_MODEL_REPOSITORY_SERVICE_CLIENT_12_003: [ The `getModel` method shall call the callback with an error parameter if a callback is passed. ]*/
+    /* Codes_SRS_NODE_MODEL_REPOSITORY_SERVICE_CLIENT_12_004: [ The `getModel` method shall return error if the method of the protocol layer failed. ]*/
+    /* Codes_SRS_NODE_MODEL_REPOSITORY_SERVICE_CLIENT_12_005: [ The `getModel` method shall return a promise if there is no callback passed. ]*/
+    return tripleValueCallbackToPromise<GetModelResponse, msRest.HttpOperationResponse, GetModelResponse>((_callback) => {
```

Comment:
so will this change to a doubleValueCallback?

Reply:
It will be simple callback

---
In reply to: [338728666](https://github.com/Azure/azure-iot-sdk-node/pull/646#discussion_r338728666) [](ancestors = 338728666)